Update estraverse

RicardoGeek opened the pull request:
I'm updating the estraverse dependency to deprecate jade, use the latest of graceful-fs and minimatch.
I think it won't break anything.

Commit:
Update estraverse

Commit:
Also updating the travisCI to use newer node LTS

RReverser commented:
Thanks for the PR, but I'm a bit confused about this:

> I'm updating the estraverse dependency to deprecate jade, use the latest of graceful-fs and minimatch.

Judging from the diff, you updated only mocha, but not estraverse (which is used here only as a peer dependency with `*` to allow usage with any version).

RicardoGeek commented:
Hi,
I'm really sorry, I didn't put a lot of thinking on my commit messages. What I meant is that I updated this module to use a new version of mocha in order to avoid the following deprecation messages while building:

```
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated graceful-fs@2.0.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
```
With the lines I changed they are all gone, and the travisCI checks passed.
